lib/Runtime/Base/ScriptContext.cpp

```diff
@@ -4122,10 +4070,11 @@ namespace Js
     {
         // Prevent reentrancy for the following work, which is not required to be done on every call to this function including
         // reentrant calls
-        if (this->isPerformingNonreentrantWork)
+        if (this->isPerformingNonreentrantWork || !this->hasInlineCache)
```

Comment:
this was only called for those scriptContexts in the zero list, from ThreadContext::ClearScriptContextCaches(), adding this to keep the same logic as before. I didn't look into the code to understand what isPerformingNonreentrantWork is though, do you think we don't need hasInlineCache here?